### ISSUE TITLE
[rush] Cannot parse PNPM version specifier for git dependency

### ISSUE THREAD

**Issue by atsikov** [microsoft/rushstack#3189]
## Summary

With `pnpm >= 6.24.0`, `rush` is not able to perform an update, failing with
```
ERROR: Cannot parse PNPM shrinkwrap version specifier:
"git@bitbucket.org+organisation/package-name/50be0b14205ba80fbdd83607830752a83c9d56a6_579d0905827311c0a8f4adac7bf6900c" for
"revolut-forms"
```

## Repro steps

1. Set "pnpmVersion" to 6.28.0 (latest at the moment of writing)
2. Add some ssh git dependency (like `ssh://git@bitbucket.org:organisation/package-name#v1.0.0`)
3. Run `rush update --full`
4. Run `rush update`

  **Expected result:** 
Rush updates dependencies according to the lock file

  **Actual result:**
Rush fails with
```
ERROR: Cannot parse PNPM shrinkwrap version specifier:
"git@bitbucket.org+organisation/package-name/50be0b14205ba80fbdd83607830752a83c9d56a6_579d0905827311c0a8f4adac7bf6900c" for
"package-name"
```

## Details

It looks like since `6.24.0`, `pnpm` changed the way "versions" for self-hosted dependencies are saved to the lock file.
For `ssh://git@bitbucket.org:organisation/package-name#v1.0.0` dependency, instead of previously used
```
package-name: bitbucket.org/organisation/package_name/hash1_hash2
```
format, it uses
```
package-name: git@bitbucket.org+organisation/package-name/hash1_hash2
```
which `rush` cannot parse.

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.61.0 |
| `rushVersion` from rush.json? | 5.61.1 |
| `useWorkspaces` from rush.json? | false |
| Operating system? | Mac |
| Would you consider contributing a PR? | No |
| Node.js version (`node -v`)? | 14.17.0 |

**Comment by iclanton** [comment]
Looks like we need to just add support for the latest version of PNPM. We should also probably emit a warning if the version of PNPM is newer than what we've tested with.

**Comment by iclanton** [comment]
I'm not able to repro this issue. Can you give a specific package specifier that I can test with? I've tried `ssh://git@github.com:webpack-contrib/json-loader.git#v0.5.5`, and it works correctly.

**Comment by atsikov** [comment]
pnpm treats github urls differently as these are kinda standard. You could reproduce the issue with some other git host, eg bitbucket. For github, dependency version is `github.com/space/package/commit-hash`, while for bitbucket it's `git@bitbucket.org+space/package/commit-hash_pnpm-hash`
You could try with this specifier: `ssh://git@bitbucket.org:atsikov-revo/rush-pnpm-test#master`

**Comment by iclanton** [comment]
PNPM isn't generating that format in the lockfile for me, but I manually tweaked it to create a repro.

**Comment by atsikov** [comment]
@iclanton thanks for the fix! Here's the repro just to double-check https://github.com/atsikov/rush-pnpm-bitbucket-dep

**Comment by iclanton** [comment]
@atsikov - the fix is published in Rush 5.62.0.
